calyx-opt/src/passes/static_promotion.rs

```diff
@@ -733,14 +997,46 @@ impl Visitor for StaticPromotion {
     fn finish_if(
         &mut self,
         s: &mut ir::If,
-        _comp: &mut ir::Component,
-        _sigs: &LibrarySignatures,
+        comp: &mut ir::Component,
+        sigs: &LibrarySignatures,
         _comps: &[ir::Component],
     ) -> VisResult {
-        if let Some(sif) = s.make_static() {
-            return Ok(Action::change(ir::Control::Static(
-                ir::StaticControl::If(sif),
-            )));
+        let mut builder = ir::Builder::new(comp, sigs);
+        if Self::can_be_promoted(&s.tbranch)
+            && Self::can_be_promoted(&s.fbranch)
+        {
+            // Both branches can be promoted
+            let approx_size_if = Self::approx_size(&s.tbranch)
```

Comment:
Annoyingly, since `visit_if` takes in an `ir::StaticIf`, we can't use the `approx_size` function, which takes in an arbitrary `Control. `